ssl/record/methods/dtls_meth.c

```diff
@@ -531,10 +642,10 @@ int dtls_get_more_records(OSSL_RECORD_LAYER *rl)
 
     if (rr->length > rl->packet_length - DTLS1_RT_HEADER_LENGTH) {
         /* now rl->packet_length == DTLS1_RT_HEADER_LENGTH */
-        more = rr->length;
-        rret = rl->funcs->read_n(rl, more, more, 1, 1, &n);
+        more = rr->length - (nread - rechdrlen);
```

Comment:
This adjustment here seems odd. We should have just set `rr->length` to the correct value in the first place, rather than trying to adjust it here.